include/alpaka/atomic/AtomicOmpBuiltIn.hpp

```diff
@@ -0,0 +1,326 @@
+/**
+* \file
+* Copyright 2014-2018 Benjamin Worpitz, Rene Widera
+*
+* This file is part of alpaka.
+*
+* alpaka is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* alpaka is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with alpaka.
+* If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#ifdef _OPENMP
+
+#include <alpaka/atomic/Traits.hpp>
+
+namespace alpaka
+{
+    namespace atomic
+    {
+        //#############################################################################
+        //! The OpenMP accelerator atomic ops.
+        //
+        //  Atomics can be used in the blocks and threads hierarchy levels.
+        //  Atomics are not guaranteed to be save between devices or grids.
+        class AtomicOmpBuiltIn
+        {
+        public:
+            using AtomicBase = AtomicOmpBuiltIn;
+
+            //-----------------------------------------------------------------------------
+            AtomicOmpBuiltIn() = default;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST AtomicOmpBuiltIn(AtomicOmpBuiltIn const &) = delete;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST AtomicOmpBuiltIn(AtomicOmpBuiltIn &&) = delete;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST auto operator=(AtomicOmpBuiltIn const &) -> AtomicOmpBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST auto operator=(AtomicOmpBuiltIn &&) -> AtomicOmpBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            /*virtual*/ ~AtomicOmpBuiltIn() = default;
```

Comment:
Not related to this PR, but to educate myself. What is the point of `/*virtual*/`?

Reply:
offline discussed wit @sbastrakov :+1: 
 - To avoid that the user is calling the destructor of this class if this class is a chield of the accelerator we can make the destructor `protected`
- An other cleaner way suggested by @sbastrakov is let the accelerator not `public` inherit all childs, instead provide a cast operator. With that way the methods of the childs will not be visible in the accelerator.

Reply:
Upon thinking a bit more about it, making the destructor `protected` should be just strictly (a little) better than the current way without any need for other changes. But  for such a base class with basically no real 'interface' inherited by accelerators, it is just not practical to do further changes.